### ISSUE TITLE
Is SQL Server Express edition supported (as instance to be monitored)?

### ISSUE THREAD

**Issue by DavidWiseman** [trimble-oss/dba-dash#74]
### Discussed in https://github.com/trimble-oss/dba-dash/discussions/71

<div type='discussions-op-text'>

<sup>Originally posted by **carehart** March 14, 2022</sup>
Quick Question: Would you say that you do or do not intend to support SQL Server Express editions? I'm having no luck connecting to a 2k8 Express edition. More below. I also don't find mention of it (either way) in the repo here.

First, nice looking too, thanks. Just learned of it today (via mention on Brent Ozar's blog post). Got it installed and connected to some servers.

But when I try to connect it to a 2008 Express edition instance, it never shows up in the tool...and your log shows a failure due to the instance not having the SQL Agent running: 
`Microsoft.SqlServer.Management.Smo.UnsupportedFeatureException: SQL Server Agent is not supported on this edition of SQL Server.`

And indeed one of the limitations of the Express edition is that the Agent is not supported (the service will not start). This is true even SQL Server 2019's implementation of Express

So do you feel dbadash does REQUIRE it, in order to connect at all? Might there be some config change I could make in the dbadash source connection that might allow it to proceed to at least connect and perhaps let is show SOME characteristics of the instance?

FWIW, I'll note that I AM indeed able to connect to the instance in question, via SSMS, from the machine on which I have implemented dbadash. And I I do run Activity Monitor from that also against that instance, using the same user I configured for dbadash to use for tha tinstance.

And I have some other SQL Server monitoring tools which work from this machine, monitoring that instance.  They all do at least connect to and show the instance, though some DO indicate how some feature of that monitoring tool may or does not work with an Express instance, sometimes specifically because of the lack of Agent support.

So bottom line, it seems that Agent support is mandatory for dbadash to connect. Or am I missing something?

Again, thanks for the tool. I will still be considering recommending it for others looking at SQL Server Monitoring solutions. And I do look forward to watching how it may evolve over time. I've also subscribed to [your youtube channel](https://www.youtube.com/channel/UCjACQ6uAfYPQJ2zydZxnAKg) to keep an eye on things.

Finally, feel free to let me know (and for the sake of other readers seeing this) whether you may prefer a question like this to be in the "issues" feature rather than in "discussions". Many github repos don't offer the "discussions" tab, so many folks never even notice it's an option to consider, when they may have something to discuss. :-)</div>

**Comment by carehart** [comment]
Thanks so much for the fix, David. Works great now for even Express 2008 (sad to admit I'm still supporting something running that). You're doing an amazing job with it all. 

I'm just dabbling with it myself to consider recommending it to others, but certainly I hope anyone who starts to rely on it would somehow contribute to your effort. To that point, I don't see an indication in the UI or this github site of your having a patreon or other means for folks to contribute. Have you considered that? I realize that being this is a Trimble-sponsored corporate project, perhaps they are not wanting to encourage outside financial contribution for some reason.

If you may want me to open this as a new issue, just say so.  If you prefer to instead just answer this "quietly" here, where someone may find it if they come looking, I'd understand. :-) Again, thanks for the efforts.

**Comment by DavidWiseman** [comment]
Great - thank you for letting me know it's working.  I've got SQL 2005 through to 2019 installed in my lab. I know some people are stuck with older versions and I'd like to keep supporting them as long as possible.  

GitHub has a sponsorship option that can be enabled which I might discuss enabling with my employer. My main motivation for open source is to allow more people to benefit from the tool.  Also for the tool to improve through open source contributions and peer review.  It's still early days for this project - you are the first person outside Trimble to contribute.  Thank you!  I still need to provide some guidance for external contributions.

**Comment by carehart** [comment]
Thanks for all that. 

I'd seen that contributing page (focused on employees), which is what clued me in to this being more of an employer-sponsored project.  And thank you to them! :-) 

And sure, I realize that it's early days and appreciate the noble open source goal of being driving by more contributors. Sadly, I've seen so many where people only take, take and never do contribute (they may not feel they can offer anything meaningful). 

And that's where some means for them to contribute monetarily may well be taken by some of them. :-) And yep, I was aware of the sponsorship feature and was alluding to that. I appreciate again that since this is an employer-sponsored site, any such option may be challenging. Just wanted to offer by it one more modest token of support for your efforts. :-)

Until again.
